hexrd/imageseries/process.py

```diff
@@ -56,9 +56,23 @@ def __iter__(self):
         return (self[i] for i in range(len(self)))
 
     def _process_frame(self, key):
-        # note: key refers to original imageseries
-        img = self._imser[key]
-        for k, d in self.oplist:
+       # note: key refers to original imageseries
+        oplist = self.oplist
+
+        # when rectangle is the first operatation we can try to call the
+        # optimized version.  If the adaptor provides one it should be
+        # signifcantly faster if not it will fallback to the same
+        # implementation that _rectangle provides.
+        if oplist and  oplist[0][0] == self.RECT:
+            region = oplist[0][1]
+            img = self._rectangle_optimized(key,region)
```

Comment:
```suggestion
        # when rectangle is the first operation we can try to call the
        # optimized version.  If the adapter provides one it should be
        # signifcantly faster if not it will fallback to the same
        # implementation that _rectangle provides.
        if oplist and  oplist[0][0] == self.RECT:
            region = oplist[0][1]
            img = self._rectangle_optimized(key, region)
```